### PR TITLE
oauth: scrubbed raw_json

### LINKED ISSUE #6299: [Oauth] sanitized raw_json when getting access_token

### PR DESCRIPTION
**Pull request by spolu**
## Description

Add support for returning a scrubbed raw_json.

We want to remove hard secrets (refresh_json) but also access_tokens to prevent users from leveraging the content of `scrubbed_raw_json` to pull access tokens.

Fixes: https://github.com/dust-tt/dust/issues/6299

## Risk

N/A

## Deploy Plan

- deploy `oauth`

### CODE REVIEW

File: connectors/src/lib/api/config.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import { EnvironmentConfig } from "@dust-tt/types";
 
 export const apiConfig = {
+  getOAuthAPIConfig: (): { url: string } => {
```
Comment (spolu) [inline connectors/src/lib/api/config.ts:4]: this one got lost but is valid to merge

File: core/src/oauth/connection.rs [hunk 2]
```diff
@@ -21,6 +21,26 @@ use tracing::error;
 
 use super::providers::confluence::ConfluenceConnectionProvider;
 
+// We hold the lock for at most 15s. In case of panic preventing the lock from being released, this
```
Comment (spolu) [inline core/src/oauth/connection.rs:24]: simple move up f the statics

File: core/src/oauth/connection.rs [hunk 3]
```diff
@@ -21,6 +21,26 @@ use tracing::error;
...
+// is the maximum time the lock will be held.
+static REDIS_LOCK_TTL_SECONDS: u64 = 15;
+// To ensure we don't write without holding the lock providers must comply to this timeout when
+// operating on tokens.
+pub static PROVIDER_TIMEOUT_SECONDS: u64 = 10;
+
+lazy_static! {
```
Comment (fontanierh) [inline core/src/oauth/connection.rs:31]: any specific reason to have 2 distinct lazy static blocks ?

**Review by fontanierh — APPROVED** [review]

File: core/src/oauth/connection.rs (hunk 3 above)
Comment (spolu) [reply core/src/oauth/connection.rs:31]: nope will fix

**Pull request merged**